python/paddle/fluid/layers/nn.py

```diff
@@ -277,6 +284,9 @@ def embedding(input,
     tmp = helper.create_tmp_variable(dtype)
     padding_idx = -1 if padding_idx is None else padding_idx if padding_idx >= 0 else (
         size[0] + padding_idx)
+    if allowed_beyond_boundary:
+        assert isinstance(candidate_idx, int) and candidate_idx >= 0, \
+            "The allowed_beyond_boundary is True, so you must set the candidate_idx."
     helper.append_op(
```

Comment:
We need also add the new attrs when appending op here.

Reply:
Done